mirrord/cli/src/main.rs

```diff
@@ -124,21 +131,49 @@ where
     );
     sub_progress_config.success(Some("config summary"));
 
+    // since execvpe doesn't exist on macOS, resolve path manually and use execve
+    let binary_path: String = if let Some(path) = var_os("PATH") {
```

Comment:
`PATH` based resolving should only happen if the given executable does not contain a `/` character.
From the execvp man page:
> The functions execlp(), execvp(), and execvP() will duplicate the actions of the shell in searching for an executable file if the specified file name does not contain a slash “/” character.  For execlp() and execvp(), search path is the path specified in the environment by “PATH” variable.  If this variable
     is not specified, the default path is set according to the _PATH_DEFPATH definition in <paths.h>, which is set to “/usr/bin:/bin”.

Also [`relibc`](https://github.com/redox-os/relibc/blob/a91cb17e4eaa82ebc807155285b3b661d8ebe0e5/src/header/unistd/mod.rs#L300) does that check.

Without that check, when executing a relative path, if that path also exists relatively to one of the paths in `PATH`, then that path would be executed, instead of the one relative to the current directory.

Regarding the last part from the docs quote - if `PATH` is not set they check relatively to /usr/bin and to /bin. Maybe you could find out what happens on shells (bash/zsh) and try to copy their behaviour. If running a certain path without mirrord works, we want it to also work with mirrord.

Reply:
sounds good to me 🫡